src/matrix.c

```diff
@@ -0,0 +1,132 @@
+#include "matrix.h"
+#include <math.h>
+
+/**
+ * @brief Converts a 2d coordinate in a matrix to an index in a 2d array
+ * @param row the row number
+ * @param col the column number
+ * @param size the number of columns in the matrix
+ */
+inline matrix_size coordToIndex(const matrix_size row, const matrix_size col,
+                                const matrix_size size) {
+  return row * size + col;
+}
+
+/**
+ * @brief Check if the given index is part of the diagonal
+ * @param index the index to check
+ * @param size the number of columns in the matrix
+ */
+inline matrix_size isDiagonal(const matrix_size index, const matrix_size size) {
+  return ((index / size) == (index % size));
+}
+
+/**
+ * @brief Multiplies 2 matrices together using this equation : output =
+ * firstMatrix x secondMatrix
+ * @param firstMatrix first matrix to multiply of size m * n
+ * @param secondMatrix second matrix to multiply of size n * p
+ * @param size array of size 3 containing m, n and p
+ * @param output matrix resulting of the matrix multiplication of the given
+ * matrices
+ * @param transposeFirstMatrix whether to transpose the first matrix when
+ * multiplying the matrices
+ */
+void matrixMultiply(const matrix_real_number* firstMatrix,
```

Comment:
This looks like a cross product. Is it one? If so, we could define this as one. Having it defined as multiply could make others confused as whether this is a dot product or a cross product.

Reply:
Cross product and dot product are only applicable to vectors. In this case we are working with matrices and those terms do not apply. I do not think such change is desirable.